modules/lang-painless/src/main/java/org/elasticsearch/painless/lookup/PainlessLookupBuilder.java

```diff
@@ -374,6 +375,9 @@ public void addPainlessClass(Class<?> clazz, Map<Class<?>, Object> annotations)
                     }
 
                     canonicalClassNamesToClasses.put(importedCanonicalClassName.intern(), clazz);
+                    if (annotations.get(AliasAnnotation.class)instanceof AliasAnnotation alias) {
+                        canonicalClassNamesToClasses.put(alias.alias(), clazz);
```

Comment:
We need validation here. This shouldn't overwrite any existing entry right?

Reply:
Added validation and test